lib/src/crypto_subtle.dart

```diff
@@ -333,31 +333,49 @@ extension type JSRsaOtherPrimesInfo(JSObject _) implements JSObject {
 }
 
 TypedData getRandomValues(TypedData array) {
+  // Since dart2wasm does not reflect values in the array, use setAll to reflect them.
+  // see: https://github.com/dart-lang/sdk/issues/59651
   if (array is Uint8List) {
-    window.crypto.getRandomValues(array.toJS);
-    return array;
+    final values = array.toJS;
+    window.crypto.getRandomValues(values);
+    if (array != values.toDart) {
+      array.setAll(0, values.toDart);
+    }
```

Comment:
Let's avoid calling `.toDart` twice, in wasm mode it'll create a wrapper, there is no need to create the wrapper twice.
```suggestion
    final jsarray = array.toJS;
    window.crypto.getRandomValues(jsarray);
    final dartArray = jsarray.toDart;
    if (array != dartArray) {
      array.setAll(0, dartArray);
    }
```

See: https://api.dart.dev/dart-js_interop/JSUint8ArrayToUint8List/toDart.html